### PR TITLE
initialcommit

### PR DESCRIPTION
**Pull request by venkateshb-ivoyant**
Added All the Functionalities

### CODE REVIEW

**Comment by shreeharibv** [comment]
Please maintain proper folder structure. Could you please modify accordingly? 
/my_spring_boot_project/
|-- src/
|-- pom.xml
|-- static/
|-- templates/
|-- application.properties
|-- test/

File: IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/CalculateTaxPayable.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        int userSecretKey = user.getSecretKey();
+        if (user == null || userSecretKey != secretKey) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+
+        UserIncome userIncomes = userIncomeRepository.findByPan(pan);
+        UserDeductions userDeductions = userDeductionsRepository.findByPan(pan);
+
+        double totalIncome = userIncomes.getOtherIncome()+userIncomes.getSalary();
+        double totalDeductions = userDeductions.getStandardDeduction()+userDeductions.getOtherDeductions()+userDeductions.getFixedDeposit()+userDeductions.getHomeLoan()+userDeductions.getHRA()+userDeductions.getInsurranceAmt();
```
Comment (shreeharibv) [inline IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/CalculateTaxPayable.java]: Please format the lines here.

File: IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+@RequestMapping("api/users/")
+public class UserController {
+    @Autowired
+    private UsersService usersService;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @PostMapping("createUser")
+    public ResponseEntity<Users> createUser(@RequestBody Users user){
```
Comment (shreeharibv) [inline IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/UserController.java]: Handle exceptions here

File: IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/UserIncomeController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    private UserIncomeRepository userIncomeRepository;
+
+    @PostMapping("createIncome/{pan}/{secretKey}")
+    public UserIncome createIncome(@PathVariable String pan, @PathVariable int secretKey, @RequestBody UserIncome userIncome) {
+        Users user = userRepository.findByPan(pan);
+        if (user.getSecretKey() == secretKey) {
+            userIncome.setPan(user.getPan());
+            return userIncomeRepository.save(userIncome);
```
Comment (shreeharibv) [inline IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/UserIncomeController.java:24]: Return proper alert messages here.

File: IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/service/UserIncomeService.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package com.ivoyant.IncomeTaxCalculator.service;
+
+public class UserIncomeService {
```
Comment (shreeharibv) [inline IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/service/UserIncomeService.java]: Delete this class if it is not used.

File: IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/service/UsersService.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+@Service
+public class UsersService {
+    @Autowired
+    private UserRepository userRepository;
+
+    public Users save(Users users){
+        if(users.getPan() != null && users.getSecretKey()!= 0){
+            return userRepository.save(users);
+        }else {
+            System.out.println("USER PAN is Empty or secret key is 0");
```
Comment (shreeharibv) [inline IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/service/UsersService.java]: Use LOGGERS instead of sysouts.

Comment (venkateshb-ivoyant) [reply IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/service/UsersService.java]: modified

File: IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/service/UserIncomeService.java (hunk 4 above)
Comment (venkateshb-ivoyant) [reply IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/service/UserIncomeService.java]: deleted & Modified

File: IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/UserIncomeController.java (hunk 3 above)
Comment (venkateshb-ivoyant) [reply IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/UserIncomeController.java:24]: Logger Added

File: IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/CalculateTaxPayable.java (hunk 1 above)
Comment (venkateshb-ivoyant) [reply IncomeTaxCalculator/IncomeTaxCalculator/src/main/java/com/ivoyant/IncomeTaxCalculator/controller/CalculateTaxPayable.java]: Formatted

**Pull request closed without merging**